### PR TITLE
[Type Checker] Fix internal error when applying unary operators to tuples with empty components

### LINKED ISSUE #8760: [Typechecker] solc ICE because of null TypePointer type
```
contract C{function f(){(int c)=++(,);}}
```

throws

https://github.com/ethereum/solidity/blob/c7b4fa34bd307f272cc7a3f6eb06e731b5629ed5/libsolidity/analysis/TypeChecker.cpp#L1088

## Environment

- Compiler version: latest develop

### PR DESCRIPTION
**Pull request by a3d4**
Fix #8760.

### CODE REVIEW

**Review by leonardoalt — CHANGES_REQUESTED** [review]
Looks good! Can you also add tests for the other unary operators?
`delete`, `-`

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -1478,13 +1478,13 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
...
 	{
-		m_errorReporter.typeError(
-			_operation.location(),
-			"Unary operator " +
-			string(TokenTraits::toString(op)) +
-			" cannot be applied to type " +
-			subExprType->toString()
-		);
+		string description = "Unary operator " + string(TokenTraits::toString(op)) + " cannot be applied to type " + subExprType->toString();
+		if (modifying)
```
Comment (chriseth) [inline libsolidity/analysis/TypeChecker.cpp:1482]: I think we should be much more liberal with applying fatal errors than we have been before.

Comment (a3d4) [reply libsolidity/analysis/TypeChecker.cpp:1482]: I could use `fatalTypeError` only, but then I would need to convert https://github.com/ethereum/solidity/blob/develop/test/libsolidity/syntaxTests/types/bool_ops.sol into several smaller tests. I would be fine with that too, no strong opinion here.

Regarding fatal vs. non-fatal errors in general. They both have pros and cons, and we need some sort of a (informal) policy which one to use in each case.

**Comment by a3d4** [comment]
@leonardoalt 
> Can you also add tests for the other unary operators?
> `delete`, `-`

Sure, 3bd1565.

File: libsolidity/analysis/TypeChecker.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/analysis/TypeChecker.cpp:1482]: The original policy was, at least for the type checker: If we cannot determine the correct type of the expression, it is a fatal error. This was relaxed with the push for type checker recovery, but as we see, it is very hard to get right.

**Review by chriseth — APPROVED** [review]

**Pull request merged**